packages/core/src/discovery.js

```diff
@@ -195,16 +223,21 @@ async function* captureSnapshotResources(page, snapshot, options) {
   };
 
   // used to resize the using capture options
-  let resizePage = width => page.resize({
-    height: snapshot.minHeight,
-    deviceScaleFactor,
-    mobile,
-    width
-  });
+  let resizePage = width => {
+    page.network.currentWidth = width;
+    return page.resize({
+      height: snapshot.minHeight,
+      deviceScaleFactor,
+      mobile,
+      width
+    });
```

Comment:
Generally if we are having too many CDP commands shouldn't we consider creating a wrapper class which will receive parameters and do the things using that?

Reply:
Actually page class is a wrapper only for running different cdp commands on browser.